php-checks/src/main/java/org/sonar/php/checks/UnusedExceptionCheck.java

```diff
@@ -39,8 +49,12 @@ public void visitExpressionStatement(ExpressionStatementTree tree) {
       if (checkForException.is(Tree.Kind.FUNCTION_CALL)) {
         checkForException = ((FunctionCallTree) checkForException).callee();
       }
-      if (((PHPTree) checkForException).getLastToken().text().endsWith("Exception")) {
-        context().newIssue(this, newExpressionTree, "Throw this exception or remove this useless statement");
+
+      if (checkForException.is(Tree.Kind.NAMESPACE_NAME)) {
+        ClassSymbol classSymbol = Symbols.getClass((NamespaceNameTree) checkForException);
+        if (classSymbol.isOrSubClassOf(EXCEPTION_FQN).isTrue() || classSymbol.isOrSubClassOf(RUNTIME_EXCEPTION_FQN).isTrue()) {
```

Comment:
As we plan to add built-in information (at least for exceptions) very soon, and RuntimeException extends Exception, I think we can remove the check fro RuntimeException here to not having to do this afterwards.

Reply:
I completely agree with you. If I remove this exam, an integration test will fail. Should I still remove RuntimeException?
(https://github.com/SonarSource/php-test-sources/blob/5b13376216ab72100466bc598976affb5514caa7/src/Symfony/Component/Process/Exception/ProcessFailedException.php)

Reply:
I would remove it.That way it is a FN for now and we will notice that we fixed it when it appears again in the tests after we add the built-in data. On the other hand, if we leave it, we will not notice that this check is useless when we add the built-in data (We could create a ticket to remember to remove it. I leave it up to you to decide :) )